### ISSUE TITLE
mapLocals logic needs work

### ISSUE THREAD

**Issue by JSStats** [WebAssembly/binaryen#118]
The logic in mapLocals looks incomplete. Not all locals appear to being added. Can be seen in the test 'fromBinary' files, and note that many of the `$var$0` instances in the output do not refer to the correct variable in the source wast files. Would be nice if missing names could have flagged an error at lookup too.

**Comment by kripken** [comment]
Can you point me to a specific local where you saw the problem? Otherwise I need to go looking at every single one in search, while you already found the problem.

**Comment by JSStats** [comment]
Many of the `$var$0` instances are bad. For example emcc_O2_hello_world.wast.fromBinary, first function `$_malloc`, about one page into the function:

emcc_O2_hello_world.wast
```
            (set_local $i3
              (i32.shr_u
                (get_local $i2)
                (i32.const 3)
              )
            )
            (set_local $i4
              (i32.load align=4
                (i32.const 176)
              )
            )
            (set_local $i5
              (i32.shr_u
                (get_local $i4)
                (get_local $i3)
              )
            )
```

emcc_O2_hello_world.wast.fromBinary 
```
            (set_local $var$0
              (i32.shr_u
                (get_local $var$1)
                (i32.const 3)
              )
            )
            (set_local $var$0
              (i32.load align=4
                (i32.const 176)
              )
            )
            (set_local $var$0
              (i32.shr_u
                (get_local $var$0)
                (get_local $var$0)
              )
            )
```

**Comment by kripken** [comment]
Thanks, now I see. Fixed in 960043a359046b1536c75aeca2f7092e15714d45
